src/merge.rs

```diff
@@ -383,9 +383,8 @@ impl Merger {
                     &mut supergraph_field.make_mut().description,
                     &field.description,
                 );
-                let mut existing_args = supergraph_field.arguments.iter();
                 for arg in field.arguments.iter() {
-                    if let Some(_existing_arg) = &existing_args.find(|a| a.name == arg.name) {
+                    if let Some(_existing_arg) = supergraph_field.argument_by_name(&arg.name) {
```

Comment:
This was subtly wrong before: by reusing the same iterator, it keeps iterating the `existing_args` iterator where it left off. It would have the wrong result if the arguments have a different order than expected. It didn't matter because the only thing depending on this is a todo comment, but still